packages/probing-eservice-operations/src/repositories/modelRepository.ts

```diff
@@ -0,0 +1,111 @@
+import { DbConfig } from "../utilities/dbConfig.js";
+import { Repository, EntityManager, DataSource } from "typeorm";
+import { FindManyOptions } from "typeorm";
+import {
+  EserviceProbingRequest,
+  EserviceProbingRequestSchema,
+} from "./entity/eservice_probing_request.entity.js";
+import {
+  EserviceProbingResponse,
+  EserviceProbingResponseSchema,
+} from "./entity/eservice_probing_response.entity.js";
+import {
+  EserviceView,
+  EserviceViewSchema,
+} from "./entity/view/eservice.entity.js";
+import { Eservice, EserviceSchema } from "./entity/eservice.entity.js";
+import { V1_DDL_1706531694434 } from "../services/db/migration/V1__DDL.js";
+import { logger } from "pagopa-interop-probing-commons";
+
+/**
+ * Extracts keys of a given type T
+ */
+type FilterKeys<T> = keyof T;
+
+/**
+ * Extracts keys of a given TypeORM entity type T
+ */
+export type TypeORMQueryKeys<T> = keyof T;
+
+/**
+ * Type of the filter that can be used to query the model.
+ * It extends the TypeORM filter type by adding all the possible model query keys.
+ * The ModelFilter includes optional filters for each property of the entity type T,
+ * and extends FindManyOptions for additional filtering options.
+ */
+export type ModelFilter<T> = {
+  [P in FilterKeys<T>]?: any; // eslint-disable-line @typescript-eslint/no-explicit-any
+} & FindManyOptions<T>;
+
+export type EserviceEntities = Repository<EserviceSchema>;
+export type EserviceProbingRequestEntities =
+  Repository<EserviceProbingRequestSchema>;
+export type EserviceProbingResponseEntities =
+  Repository<EserviceProbingResponseSchema>;
+export type EserviceViewEntities = Repository<EserviceViewSchema>;
+
+export class ModelRepository {
+  private static instance: ModelRepository;
+
+  private connection: DataSource;
+  private entityManager: EntityManager;
+
+  public eservices: EserviceEntities;
+  public eserviceProbingRequest: EserviceProbingRequestEntities;
+  public eserviceProbingResponse: EserviceProbingResponseEntities;
+  public eserviceView: EserviceViewEntities;
+
+  private constructor({
+    dbHost: host,
+    dbPort: port,
+    dbUsername: username,
+    dbPassword: password,
+    dbName: database,
+  }: DbConfig) {
+    this.connection = new DataSource({
+      type: "postgres",
+      name: "probing-eservice-operations",
+      host,
+      port,
+      username,
+      password,
+      database,
+      entities: [
+        Eservice,
+        EserviceProbingRequest,
+        EserviceProbingResponse,
+        EserviceView,
+      ],
+      migrations: [V1_DDL_1706531694434],
```

Comment:
is it possible to use a folder of sql files instead?
if not, I would think about performing the migrations outside the application